### PR TITLE
Use mock of IProjectPropertyManager rather than an actual instance in ProjectTestPropertySetCommandTests

### PR DESCRIPTION
**Pull request by bigfluffycookie**
part of https://github.com/SonarSource/sonarlint-visualstudio/issues/4770

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs [hunk 1]
```diff
@@ -19,9 +19,9 @@
  */
```
Comment (bigfluffycookie) [inline src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs:19]: Lots of tests in here are identical to the one in the PR -> https://github.com/SonarSource/sonarlint-visualstudio/pull/4828

I'll leave a pr comment wherever this applies

File: src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs [hunk 2]
```diff
@@ -33,150 +33,57 @@ namespace SonarLint.VisualStudio.Integration.UnitTests.Commands
...
+            act = () => new ProjectTestPropertySetCommand(Mock.Of<IProjectPropertyManager>(), null, true);
             act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("projectToLanguageMapper");
         }
 
         [TestMethod]
-        public void ProjectTestPropertySetCommand_Invoke_SingleProject_SetsValue()
+        [DataRow(true)]
+        [DataRow(false)]
+        [DataRow(null)]
+        public void Invoke_MultipleProjects_SetsValues(bool? setPropertyValue)
```
Comment (bigfluffycookie) [inline src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs:52]: There was no real reason to test single project and multiple projects for this specific method so I removed the SingleProject one.

File: src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs [hunk 3]
```diff
@@ -33,150 +33,57 @@ namespace SonarLint.VisualStudio.Integration.UnitTests.Commands
...
             // Assert
-            this.VerifyTestProperty(project, null);
+            propertyManager.Verify(x => x.SetBooleanProperty(p1, "SonarQubeTestProject", setPropertyValue), Times.Once);
+            propertyManager.Verify(x => x.SetBooleanProperty(p2, "SonarQubeTestProject", setPropertyValue), Times.Once);
+            propertyManager.Verify(x => x.SetBooleanProperty(p3, "SonarQubeTestProject", setPropertyValue), Times.Once);
         }
 
         [TestMethod]
-        public void ProjectTestPropertySetCommand_Invoke_MultipleProjects_MixedPropValues_SetsValues()
+        public void QueryStatus_MissingPropertyManager_IsDisabledIsHidden()
```
Comment (bigfluffycookie) [inline src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs:81]: Identical to the method with the same name in the other pr

File: src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs [hunk 4]
```diff
@@ -187,319 +94,135 @@ public void ProjectTestPropertySetCommand_QueryStatus_MissingPropertyManager_IsD
         }
 
         [TestMethod]
-        public void ProjectTestPropertySetCommand_QueryStatus_SingleProject_SupportedProject_IsEnabledIsVisible()
+        [DataRow(true)]
+        [DataRow(false)]
+        public void QueryStatus_SingleProject_Project_Supported_Unsupported_CorrectCommandState(bool projectSupported)
```
Comment (bigfluffycookie) [inline src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs:99]: Identical to the method with the same name in the other pr

File: src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs [hunk 5]
```diff
@@ -187,319 +94,135 @@ public void ProjectTestPropertySetCommand_QueryStatus_MissingPropertyManager_IsD
...
-            command.Visible.Should().BeTrue("Expected command to be visible");
+            command.Enabled.Should().IsSameOrEqualTo(projectSupported);
+            command.Visible.Should().IsSameOrEqualTo(projectSupported);
         }
 
         [TestMethod]
-        public void ProjectTestPropertySetCommand_QueryStatus_SingleProject_UnsupportedProject_IsDisabledIsHidden()
+        [DataRow(true)]
+        [DataRow(false)]
+        public void QueryStatus_SingleProject_CheckedStateReflectsValues(bool excludeValue)
```
Comment (bigfluffycookie) [inline src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs:124]: Identical to the method with the same name in the other pr

File: src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs [hunk 6]
```diff
@@ -187,319 +94,135 @@ public void ProjectTestPropertySetCommand_QueryStatus_MissingPropertyManager_IsD
...
+            command.Checked.Should().IsSameOrEqualTo(excludeValue);
         }
 
         [TestMethod]
-        public void ProjectTestPropertySetCommand_QueryStatus_MultipleProjects_MixedPropValues_IsUnchecked()
+        [DataRow(true, true, true)]
+        [DataRow(false, false, false)]
+        [DataRow(null, false, false)]
+        [DataRow(null, true, false)]
+        public void QueryStatus_MultipleProjects_CheckedStateReflectsValues(bool? excludeValue1, bool? excludeValue2, bool expectedCheckedValue)
```
Comment (bigfluffycookie) [inline src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs:151]: Identical to the method with the same name in the other pr

**Review by duncanp-sonar — APPROVED** [review]
NIce - much simpler!

**Pull request merged**